funds-manager/funds-manager-server/src/custody_client/gas_wallets.rs

```diff
@@ -59,8 +94,9 @@ impl CustodyClient {
         let secret_name = Self::gas_wallet_secret_name(&gas_wallet.address);
         let secret_value = get_secret(&secret_name, &self.aws_config).await?;
 
-        // Update the gas wallet to be active and return the keypair
+        // Update the gas wallet to be active, top up wallets, and return the key
         self.mark_gas_wallet_active(&gas_wallet.address, peer_id).await?;
+        self.refill_gas_for_active_wallets(DEFAULT_TOP_UP_AMOUNT).await?;
```

Comment:
why do we top up all the active gas wallets here, as opposed to just the new one?

Reply:
simpler, i see no reason not to